pkg/render/applicationlayer/applicationlayer.go

```diff
@@ -214,13 +244,46 @@ func (c *component) containers() []corev1.Container {
 	}
 	containers = append(containers, proxy)
 
-	collector := corev1.Container{
-		Name:         L7CollectorContainerName,
-		Image:        c.config.collectorImage,
-		Env:          c.collectorEnv(),
-		VolumeMounts: c.collectorVolMounts(),
+	if c.config.LogsEnabled {
+		// Log collection specific container
+		collector := corev1.Container{
+			Name:         L7CollectorContainerName,
+			Image:        c.config.collectorImage,
+			Env:          c.collectorEnv(),
+			VolumeMounts: c.collectorVolMounts(),
+		}
+		containers = append(containers, collector)
+	}
+
+	if c.config.WafEnabled {
+		// Web Application Firewall (WAF) specific container
+		dikastes := corev1.Container{
+			Name:  DikastesContainerName,
+			Image: c.config.dikastesImage,
+			Command: []string{
+				"/dikastes",
+				"server",
+				"--dial", "/var/run/felix/nodeagent/socket",
+				"--listen", "/var/run/dikastes/dikastes.sock",
+				"--rules", "/etc/modsecurity-ruleset",
+			},
+			Env: []corev1.EnvVar{
+				{Name: "LOG_LEVEL", Value: "Info"},
+			},
+			VolumeMounts: []corev1.VolumeMount{
+				{Name: FelixSync, MountPath: "/var/run/felix"},
+				{Name: DikastesSyncVolumeName, MountPath: "/var/run/dikastes"},
+				{Name: CalicoLogsVolumeName, MountPath: "/var/log/calico"},
+				{Name: ModSecurityRulesetVolumeName, MountPath: "/etc/modsecurity-ruleset", ReadOnly: true},
+			},
+			SecurityContext: &corev1.SecurityContext{
```

Comment:
Have we considered at all if there is a way around this requirement? We really shouldn't be running as root unless absolutely necessary. Talk with @mgleung about options here. This is another one to add to the backlog, because we likely won't be able to address in this release but we should fix it next release.

Reply:
Backlog item it is. Thanks!